src/orchestrator-process/src/lib.rs

```diff
@@ -525,6 +527,22 @@ impl NamespacedProcessOrchestrator {
                 }
             }
 
+            // Clean up scratch directory when the service is terminated.
+            // This is best effort as a development and testing convenience.
+            // Because the process orchestrator is not used in production, we
+            // don't need to be perfectly robust with the cleanup.
+            let _guard = scopeguard::guard((), |_| {
+                if let Some(scratch) = scratch_directory {
+                    info!(scratch_dir = %scratch.display(), "cleaning up scratch directory");
+                    if let Err(e) = std::fs::remove_dir_all(scratch) {
```

Comment:
we shouldn't be doing blocking io on a tokio thread, this could really slow down things in production

There are no `?`'s or early returns (other than panic's, which abort the process anyways) after this, so I think you can use `tokio::fs::remove_dir_all` just after the `loop`

Reply:
Made the same comment here: https://github.com/MaterializeInc/materialize/pull/19761#discussion_r1220856259

Putting this below the `loop` won't work though; that'll be dead code. The loop only terminates when the future is dropped.

Reply:
Updated to spawn a task and use `tokio::fs::remove_dir_all`

Reply:
I did push. For some reason I think github is lagging

Reply:
Commit finally showed up!

Reply:
Ah right, GitHub outage: https://www.githubstatus.com/incidents/1g1gkh0qpyvs